development/extensions/skeleton_extension.rst

```diff
@@ -0,0 +1,750 @@
+to configure and generate a skeleton that will suit your needs, so you
+can jump straight into writing your own code.
+
+Extension packaging
+-------------------
+
+Extensions are packaged in a directory structure as follows: ``vendorname/extensionname``.
+For example, the Acme Demo extension uses a packaging structure that looks like ``acme/demo``.
+
+Vendor name
+^^^^^^^^^^^
+
+The vendor name can be your username or any other name you choose to associate 
+your extensions with. Valid naming conventions must start with a letter and
+contain letters and numbers only. In the case of the Acme Demo extension, 
+this would be ``acme``. **This field is required.**
+
+Package name
+^^^^^^^^^^^^
+
+The package name is the name of the extension. Valid naming conventions must
+start with a letter and contain letters and numbers only. In the case of the 
+Acme Demo extension, this would be ``demo``. **This field is required.**
+
+Display name
+^^^^^^^^^^^^
+
+This is the name (or title) of your extension in plain English. This is what the user will
+see when your extension appears in the phpBB ACP Extension Manager. In the case of the 
+Acme Demo extension, this would be ``Acme Demo``. **This field is required.**
+
+Description
+^^^^^^^^^^^
+
+A brief description of your extension. This will be visible to users in the Details page
+of your extension in phpBB's ACP Extension Manager.
+
+Version
+^^^^^^^
+
+The version of your extension. phpBB's coding guidelines require an ``x.y.z`` semantic 
+versioning format, with optional ``-dev``, ``-alpha``, ``-beta`` and ``-RC`` pre-release suffixes.
+When left blank, the Skeleton Extension will use ``1.0.0-dev`` by default.
+
+Date of the extension
+^^^^^^^^^^^^^^^^^^^^^
+
+Date the extension was created. Dates must be in the ``YYYY-MM-DD`` format. 
+When left blank, the Skeleton Extension will use the current date by default.
+
+Homepage of the extension
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An optional link to your extension's homepage. This can be, for example, an extension's
+GitHub repository or any other site related to your extension.
+
+Authors
+-------
+
+Here you can name your extension's author(s). Use the ``Add author`` button
+to add additional authors.
+
+Author name
+^^^^^^^^^^^
+
+The name of an extension's author. Must not be empty.
+
+Author email
+^^^^^^^^^^^^
+
+The email address of an extension's author. (Optional)
+
+Author homepage
+^^^^^^^^^^^^^^^
+
+The homepage URL of an extension's author. (Optional)
+
+Author role
+^^^^^^^^^^^
+
+The role of an extension's author. For example, lead developer, contributor, 
+translator, etc. (Optional)
+
+Requirements
+------------
+
+Requirements are where you define the minimum (and maximum) PHP and phpBB versions
+your extension needs to be able to perform.
+
+Minimum phpBB requirement of the extension
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Typically this should be set to the version of phpBB in which you are developing
```

Comment:
Not really specific to the documentation (But might be mentioned here anyway), that the minimum version is not checked/enforced by phpBB currently. Also might be a good idea, if it is not happening already, to generate a ext.php that matches this (And the previous check) check.